### PR TITLE
Fix sql query parameterization bug using $1, $2 placeholders

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes
Gets this $1, $2 syntax working for relational databases, as well as mongodb.

@cw75 as lead reviewer. @likawind to take a look at MongoDB and any missing coverage there.

UI follow-up task is tracked here: https://linear.app/aqueducthq/issue/ENG-2693/show-an-executed-query-and-query-details-tab-in-the-extra-operator

## Related issue number (if any)
ENG-2674

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Review by likawind — COMMENTED** [review]
Looks like we are keeping the current bracket behavior. In this case, should we add test cases instead of modifying existing one?

**Comment by kenxu95** [comment]
We're not keeping the current bracket behavior. Bracketing only works for "builtin"-functions ike today.

File: src/python/aqueduct_executor/operators/connectors/data/execute.py [hunk 1]
```diff
@@ -142,9 +141,7 @@ def run_extract(
         assert all(
             isinstance(param_val, str) for param_val in input_vals
         ), "Parameter value must be a string."
-
-        parameters = dict(zip(spec.input_param_names, input_vals))
-        extract_params.compile(parameters)
+        extract_params.compile(input_vals)
```
Comment (cw75) [inline src/python/aqueduct_executor/operators/connectors/data/execute.py:144]: we don't need this step anymore?

**Review by cw75 — APPROVED** [review]
Left a couple minor comments otherwise LGTM!

File: src/python/aqueduct_executor/operators/connectors/data/execute.py (hunk 1 above)
Comment (kenxu95) [reply src/python/aqueduct_executor/operators/connectors/data/execute.py:144]: ah yeah - we don't need it cus we only need a list of input parameter values, which `input_vals` already is.

File: sdk/aqueduct/integrations/mongodb_integration.py [hunk 2]
```diff
@@ -57,6 +57,29 @@ def find(
...
             output:
                 Name to assign the output artifact. If not set, the default naming scheme will be used.
+            parameters:
+                An optional list of string parameters to use in the query.  We use the Postgres syntax of $1, $2 for placeholders.
+                The number denotes which parameter in the list to use (one-indexed). These parameters feed into the
+                sql query operator and will fill in the placeholders in the query with the actual values.
+
+                Example:
+                    country_param = client.create_param("UK", default=" United Kingdom ")
+                    empty_param = client.create_param("Thailand", default=" Thailand ")
```
[the remark this answers is not in the document]
Comment (kenxu95) [reply sdk/aqueduct/integrations/mongodb_integration.py:67]: leftover from a namign change. will fix

**Pull request merged**